### PR TITLE
Add DigitalOcean automatic DNS API support (native)

### PR DESCRIPTION
**Pull request by gitwer**
Added DigitalOcean native API support for automatic dns validation.

### CODE REVIEW

File: dnsapi/dns_digitalocean.sh [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    _err "invalid domain or split"
+    return 1
+  fi
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  ## Set the header with our post type and key auth key
+  export _H1="Content-Type: application/json"
+  export _H2="Authorization: Bearer $DO_API_KEY"
+  PURL='https://api.digitalocean.com/v2/domains/'$_domain'/records'
```
Comment (Neilpang) [inline dnsapi/dns_digitalocean.sh]: Please call `_saveaccountconf DO_API_KEY "$DO_API_KEY"` to save the key for future automatic renewal.

File: dnsapi/dns_digitalocean.sh [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  ## the create request - post
+  ## args: BODY, URL, [need64, httpmethod]
+  response="$(_post "$PBODY" "$PURL")"
+
+  ## check response (sort of)
+  if [ "$?" != "0" ]; then
+    _err "error in response: $response"
+    return 1
+  fi
+  _debug response "$response"
```
Comment (Neilpang) [inline dnsapi/dns_digitalocean.sh]: use `_debug2` for the response.

File: dnsapi/dns_digitalocean.sh [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    record="$(echo "$domain_list" | _egrep_o "\"id\"\s*\:\s*\"*\d+\"*[^}]*\"name\"\s*\:\s*\"$_sub_domain\"[^}]*\"data\"\s*\:\s*\"$txtvalue\"")"
+    ## 3) check record and get next page
+    if [ -z "$record" ]; then
+      ## find the next page if we dont have a match
+      nextpage="$(echo "$domain_list" | _egrep_o "\"links\".*" | _egrep_o "\"next\".*" | _egrep_o "http.*page\=\d+")"
+      if [ -z "$nextpage" ]; then
+        _err "no record and no nextpage in digital ocean DNS removal"
+        return 1
+      fi
+      _debug nextpage "$nextpage"
```
Comment (Neilpang) [inline dnsapi/dns_digitalocean.sh]: use `_debug2` here

File: dnsapi/dns_digitalocean.sh [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  ## the create request - delete
+  ## args: BODY, URL, [need64, httpmethod]
+  response="$(_post "" "$DURL" "" "DELETE")"
+
+  ## check response (sort of)
+  if [ "$?" != "0" ]; then
+    _err "error in remove response: $response"
+    return 1
+  fi
+  _debug response "$response"
```
Comment (Neilpang) [inline dnsapi/dns_digitalocean.sh]: `_debug2`

**Comment by Neilpang** [comment]
Do you think `dns_digitalocean` is a too long name ?

how about `dns_dgon` ?

**Comment by Neilpang** [comment]
Please update the usage to this readme file too:

https://github.com/Neilpang/acme.sh/blob/master/dnsapi/README.md

**Comment by gitwer** [comment]
Just realised I have merged your dev into my forked master and am requesting a pull back to your dev. Not sure if that will cause a problem (first time using any sort of branching, forking, pull requests etc). Let me know if you want me to close this request and try again, my forked dev (with updates) to your dev.

**Comment by Neilpang** [comment]
Yes,  please close this PR and send me a new one.

**Pull request closed without merging**